nixos/modules/services/networking/vpp.nix

```diff
@@ -0,0 +1,543 @@
+          description = ''
+            Configuration file passed to {command}`vpp -c`.
+            It is recommended to use the {option}`settings` option instead.
+
+            Setting this option will override the config file
+            auto-generated from the {option}`settings` option.
+          '';
+        };
+
+        startupConfig = mkOption {
+          type = types.str;
+          default = "";
+          example = ''
+            set interface state TenGigabitEthernet1/0/0 up
+            set interface state TenGigabitEthernet1/0/1 up
+            set interface ip address TenGigabitEthernet1/0/0 2001:db8::1/64
+            set interface ip address TenGigabitEthernet1/0/1 2001:db8:1234::1/64
+          '';
+          description = ''
+            Script to run on startup in {command}`vppctl`, passed to
+            VPP's `startup-config` option in the `unix` section as a file.
+
+            This is used to configure things like IP addresses and routes. To configure
+            interfaces, plugins, etc., see the {option}`settings` option.
+          '';
+        };
+
+        startupConfigFile = mkOption {
+          type = types.path;
+          example = "./vpp-startup.conf";
+          default = pkgs.writeText "vpp-${name}-startup.conf" cfg.instances.${name}.startupConfig;
+          defaultText = ''
+            pkgs.writeText "vpp-\${name}-startup.conf" (builtins.toString config.services.vpp.instances.\${name}.startupConfig);
+          '';
+          description = ''
+            File to run as a script on startup in {command}`vppctl`, passed
+            to VPP's `startup-config` option in the `unix` section.
+
+            Setting this option will override {option}`startupConfig`.
+          '';
+        };
+      };
+    };
+in
+{
+  options.services.vpp = {
+    hugepages = {
+      autoSetup = mkOption {
+        default = false;
+        example = true;
+        description = ''
+          Whether to automatically setup hugepages for use with FD.io's Vector Packet Processor.
+
+          If any programs other than VPP use hugepages or you want to use 1GB hugepages, it's recommended
+          to keep this `false` and set them up manually using {option}`boot.kernel.sysctl` or {option}`boot.kernelParams`.
+        '';
+      };
+
+      count = mkOption {
+        type = types.ints.positive;
+        default = 1024;
+        example = 512;
+        description = "Number of 2MB hugepages to setup on the system if {option}`services.vpp.hugepages.autoSetup` is enabled.";
+      };
+    };
+
+    instances = mkOption {
+      default = { };
+      type = types.attrsOf (types.submodule vppOpts);
+      description = ''
+        VPP supports multiple instances for testing or other purposes.
+        If you don't require multiple instances of VPP you can define just the one.
+      '';
+      example = {
+        main = {
+          enable = true;
+          group = "vpp-main";
+          settings = { };
+        };
```

Comment:
Two things about the instance name here - this is more of a personal preference, but I find it good practice to quote any arbitrary `<name>` sections like
```nix
services.vpp.instances."main" = {...};
```
I find it helps greatly with code readability, especially since most nix syntax highlighters make quoted text a different color.

When it comes to `example` here I think it might be a trade off, since while looking at the source its quoted and highlighted, but I'm not sure if that shows up in the docs unless you use `literalExpression` - either it's easier to read the source, or it shows up in the docs for the end user. I could be wrong about that rendering though.


Second thing, you seem to use `<name>` pretty much everywhere except on line 114, where you have it as `<instances>` - I would just make sure it's consistent across the whole module.